### DISCUSSION TITLE
encrypt public key before saving it in the database

### DISCUSSION THREAD

**Discussion by mustafasalfiti** [eclipse-tractusx/managed-identity-wallet#175] [Q&A]
when i was going through the code i was wondering why are we encrypting the public key for a wallet before saving it in the database?

shouldn't the public key be in our case visible?

```
        //Save key
        walletKeyService.getRepository().save(WalletKey.builder()
                .walletId(wallet.getId())
                .keyId(keyId)
                .referenceKey("dummy ref key, removed once vault setup is ready")
                .vaultAccessToken("dummy vault access token, removed once vault setup is ready")
                .privateKey(encryptionUtils.encrypt(getPrivateKeyString(keyPair.getPrivateKey().asByte()))) 
                .publicKey(encryptionUtils.encrypt(getPublicKeyString(keyPair.getPublicKey().asByte()))) // why?
                .build());
        log.debug("Wallet created for bpn ->{}", StringEscapeUtils.escapeJava(request.getBpn()));
```

**Comment by m-gora** [comment]
good question indeed. it's a PUBLIC key after all.

**Reply by borisrizov-zf** [reply]
I think someone in the past meant to encrypt the private key, but no way to know for sure.

There is a concept to use Vault for key storage, which would deprecate this code and resolve the discussion.
If we decide to fix it until a Vault solution is available, we can do so.

**Comment by nitin-vavdiya** [comment]
There is no such concrete reason for encrypting the public key before storing it.
We are not retrieving the public key from the database once it is stored, We are using did document for credential verification.

We can remove it
